llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1082,13 +1047,9 @@ class VPInstructionWithType : public VPInstruction {
 
 public:
   VPInstructionWithType(unsigned Opcode, ArrayRef<VPValue *> Operands,
-                        Type *ResultTy, DebugLoc DL, const Twine &Name = "")
-      : VPInstruction(Opcode, Operands, DL, Name), ResultTy(ResultTy) {}
-  VPInstructionWithType(unsigned Opcode,
-                        std::initializer_list<VPValue *> Operands,
-                        Type *ResultTy, FastMathFlags FMFs, DebugLoc DL = {},
+                        Type *ResultTy, const VPIRFlags &Flags, DebugLoc DL,
                         const Twine &Name = "")
-      : VPInstruction(Opcode, Operands, FMFs, DL, Name), ResultTy(ResultTy) {}
+      : VPInstruction(Opcode, Operands, Flags, DL, Name), ResultTy(ResultTy) {}
```

Comment:
Should we keep the constructor with no VPIRFlags to keep it in sync with VPInstruction?

Or alternatively make it a default argument

Reply:
The reason I removed the constructor here but not VPInstruction is that  VPINstructionWithTypes are all created thorugh VPBuilder, so there are very few places that need updating, which is different for VPInstruction, where we would need to pull in quite a few additional updates. We could move the Flags argument to the end, but that position would seem inconsistent I think.